src/providers/hana/qgshananewconnection.cpp

```diff
@@ -0,0 +1,358 @@
+/***************************************************************************
+   qgshananewconnection.cpp
+   --------------------------------------
+   Date      : 31-05-2019
+   Copyright : (C) SAP SE
+   Author    : Maksim Rylov
+ ***************************************************************************/
+
+/***************************************************************************
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ ***************************************************************************/
+#include "qgsauthmanager.h"
+#include "qgshanaconnection.h"
+#include "qgshananewconnection.h"
+#include "qgshanasettings.h"
+#include "qgssettings.h"
+
+#include <QInputDialog>
+#include <QMessageBox>
+
+using namespace std;
+
+namespace
+{
+  bool isStringEmpty( const QString &input )
+  {
+    return ( input.isEmpty() || QString( input ).replace( " ", "" ).isEmpty() );
+  }
+}
+
+QgsHanaNewConnection::QgsHanaNewConnection(
+  QWidget *parent,
+  const QString &connName,
+  Qt::WindowFlags fl )
+  : QDialog( parent, fl )
+  , mOriginalConnName( connName )
+{
+  setupUi( this );
+
+  cmbIdentifierType_changed( cmbIdentifierType->currentIndex() );
+
+  connect( cmbIdentifierType, static_cast<void ( QComboBox::* )( int )>( &QComboBox::currentIndexChanged ), this, &QgsHanaNewConnection::cmbIdentifierType_changed );
+  connect( rbtnSingleContainer, &QRadioButton::clicked, this, &QgsHanaNewConnection::rbtnSingleContainer_clicked );
+  connect( rbtnMultipleContainers, &QRadioButton::clicked, this, &QgsHanaNewConnection::rbtnMultipleContainers_clicked );
+  connect( rbtnTenantDatabase, &QRadioButton::clicked, this, &QgsHanaNewConnection::rbtnTenantDatabase_clicked );
+  connect( rbtnSystemDatabase, &QRadioButton::clicked, this, &QgsHanaNewConnection::rbtnSystemDatabase_clicked );
+  connect( chkEnableSSL, &QCheckBox::clicked, this, &QgsHanaNewConnection::chkEnableSSL_clicked );
+  connect( chkValidateCertificate, &QCheckBox::clicked, this, &QgsHanaNewConnection::chkValidateCertificate_clicked );
+  connect( btnConnect, &QPushButton::clicked, this, &QgsHanaNewConnection::btnConnect_clicked );
+  connect( buttonBox, &QDialogButtonBox::helpRequested, this, &QgsHanaNewConnection::showHelp );
+
+#ifdef Q_OS_WIN
+  txtDriver->setText( QLatin1String( "HDBODBC" ) );
+#else
+  txtDriver->setText( QLatin1String( "/usr/sap/hdbclient/libodbcHDB.so" ) );
```

Comment:
looks Linux specific, but according to https://github.com/qgis/QGIS-Enhancement-Proposals/issues/151#issuecomment-533136222, MacOS support is supposed to be available

Reply:
This issue has been resolved in f6aadcf.